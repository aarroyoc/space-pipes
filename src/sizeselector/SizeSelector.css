.sizeselector{
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    height: 100%;
}

.sizeselector__buttons{
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    width: 100%;
    margin-top: 1.5em;
    margin-bottom: 1.5em;
}

.home__padding{
    height: 0.5em;
}