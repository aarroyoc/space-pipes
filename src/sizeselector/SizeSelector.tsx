import React from 'react';
import { useHistory } from "react-router-dom";
// @ts-ignore
import { Heading, Button } from 'arwes';

import "./SizeSelector.css";

export default function SizeSelector(){
    const history = useHistory();
    return (
            <div className="home">
                <Heading node="h1">SpacePipes</Heading>
                <p>Select the size of the puzzle</p>
                <div className="home__buttons">
                    <Button animate onClick={() => history.push("/netwalk/5")}>5x5</Button>
                    <div className="home__padding"></div>
                    <Button animate onClick={() => history.push("/netwalk/7")}>7x7</Button>
                    <div className="home__padding"></div>
                    <Button animate onClick={() => history.push("/netwalk/9")}>9x9</Button>
                </div>
            </div>
    );
}