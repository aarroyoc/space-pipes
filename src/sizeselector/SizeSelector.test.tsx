import React from 'react';
import { render } from '@testing-library/react';
// @ts-ignore
import { ThemeProvider, createTheme } from 'arwes';
import SizeSelector from "./SizeSelector";

test("render matchs snapshot", () => {
    const {container} = render(
        <ThemeProvider theme={createTheme()}>
            <SizeSelector/>
        </ThemeProvider>);
    expect(container).toMatchSnapshot();
});