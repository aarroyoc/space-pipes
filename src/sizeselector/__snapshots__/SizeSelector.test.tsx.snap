// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matchs snapshot 1`] = `
<div>
  <div
    class="home"
  >
    <h1
      class="Heading-root-0-1-1"
    >
      SpacePipes
    </h1>
    <p>
      Select the size of the puzzle
    </p>
    <div
      class="home__buttons"
    >
      <div
        class="ArwesSounds(Button)-root-0-1-2"
      >
        <div
          class="ArwesSounds(Frame)-root-0-1-5 ArwesSounds(Frame)-root-0-1-23 ArwesSounds(Frame)-hover-0-1-18 ArwesSounds(Frame)-entering-0-1-19"
        >
          <div
            class="ArwesSounds(Frame)-border-0-1-8 ArwesSounds(Frame)-border-0-1-25 ArwesSounds(Frame)-borderLeft-0-1-9"
          />
          <div
            class="ArwesSounds(Frame)-border-0-1-8 ArwesSounds(Frame)-border-0-1-25 ArwesSounds(Frame)-borderRight-0-1-10"
          />
          <div
            class="ArwesSounds(Frame)-border-0-1-8 ArwesSounds(Frame)-border-0-1-25 ArwesSounds(Frame)-borderTop-0-1-11"
          />
          <div
            class="ArwesSounds(Frame)-border-0-1-8 ArwesSounds(Frame)-border-0-1-25 ArwesSounds(Frame)-borderBottom-0-1-12"
          />
          <div
            class="ArwesSounds(Frame)-corner-0-1-13 ArwesSounds(Frame)-corner-0-1-26 ArwesSounds(Frame)-cornerLT-0-1-14 ArwesSounds(Frame)-cornerLT-0-1-27"
          />
          <div
            class="ArwesSounds(Frame)-corner-0-1-13 ArwesSounds(Frame)-corner-0-1-26 ArwesSounds(Frame)-cornerLB-0-1-15 ArwesSounds(Frame)-cornerLB-0-1-28"
          />
          <div
            class="ArwesSounds(Frame)-corner-0-1-13 ArwesSounds(Frame)-corner-0-1-26 ArwesSounds(Frame)-cornerRT-0-1-16 ArwesSounds(Frame)-cornerRT-0-1-29"
          />
          <div
            class="ArwesSounds(Frame)-corner-0-1-13 ArwesSounds(Frame)-corner-0-1-26 ArwesSounds(Frame)-cornerRB-0-1-17 ArwesSounds(Frame)-cornerRB-0-1-30"
          />
          <div
            class="ArwesSounds(Frame)-box-0-1-6 ArwesSounds(Frame)-box-0-1-24"
          >
            <div
              class="ArwesSounds(Frame)-children-0-1-7"
            >
              <div
                class="Highlight-root-0-1-31"
              >
                <div
                  class="Highlight-children-0-1-33"
                >
                  <button
                    class="ArwesSounds(Button)-button-0-1-3 ArwesSounds(Button)-button-0-1-4"
                  >
                    5x5
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="home__padding"
      />
      <div
        class="ArwesSounds(Button)-root-0-1-2"
      >
        <div
          class="ArwesSounds(Frame)-root-0-1-5 ArwesSounds(Frame)-root-0-1-36 ArwesSounds(Frame)-hover-0-1-18 ArwesSounds(Frame)-entering-0-1-19"
        >
          <div
            class="ArwesSounds(Frame)-border-0-1-8 ArwesSounds(Frame)-border-0-1-38 ArwesSounds(Frame)-borderLeft-0-1-9"
          />
          <div
            class="ArwesSounds(Frame)-border-0-1-8 ArwesSounds(Frame)-border-0-1-38 ArwesSounds(Frame)-borderRight-0-1-10"
          />
          <div
            class="ArwesSounds(Frame)-border-0-1-8 ArwesSounds(Frame)-border-0-1-38 ArwesSounds(Frame)-borderTop-0-1-11"
          />
          <div
            class="ArwesSounds(Frame)-border-0-1-8 ArwesSounds(Frame)-border-0-1-38 ArwesSounds(Frame)-borderBottom-0-1-12"
          />
          <div
            class="ArwesSounds(Frame)-corner-0-1-13 ArwesSounds(Frame)-corner-0-1-39 ArwesSounds(Frame)-cornerLT-0-1-14 ArwesSounds(Frame)-cornerLT-0-1-40"
          />
          <div
            class="ArwesSounds(Frame)-corner-0-1-13 ArwesSounds(Frame)-corner-0-1-39 ArwesSounds(Frame)-cornerLB-0-1-15 ArwesSounds(Frame)-cornerLB-0-1-41"
          />
          <div
            class="ArwesSounds(Frame)-corner-0-1-13 ArwesSounds(Frame)-corner-0-1-39 ArwesSounds(Frame)-cornerRT-0-1-16 ArwesSounds(Frame)-cornerRT-0-1-42"
          />
          <div
            class="ArwesSounds(Frame)-corner-0-1-13 ArwesSounds(Frame)-corner-0-1-39 ArwesSounds(Frame)-cornerRB-0-1-17 ArwesSounds(Frame)-cornerRB-0-1-43"
          />
          <div
            class="ArwesSounds(Frame)-box-0-1-6 ArwesSounds(Frame)-box-0-1-37"
          >
            <div
              class="ArwesSounds(Frame)-children-0-1-7"
            >
              <div
                class="Highlight-root-0-1-31"
              >
                <div
                  class="Highlight-children-0-1-33"
                >
                  <button
                    class="ArwesSounds(Button)-button-0-1-3 ArwesSounds(Button)-button-0-1-35"
                  >
                    7x7
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="home__padding"
      />
      <div
        class="ArwesSounds(Button)-root-0-1-2"
      >
        <div
          class="ArwesSounds(Frame)-root-0-1-5 ArwesSounds(Frame)-root-0-1-46 ArwesSounds(Frame)-hover-0-1-18 ArwesSounds(Frame)-entering-0-1-19"
        >
          <div
            class="ArwesSounds(Frame)-border-0-1-8 ArwesSounds(Frame)-border-0-1-48 ArwesSounds(Frame)-borderLeft-0-1-9"
          />
          <div
            class="ArwesSounds(Frame)-border-0-1-8 ArwesSounds(Frame)-border-0-1-48 ArwesSounds(Frame)-borderRight-0-1-10"
          />
          <div
            class="ArwesSounds(Frame)-border-0-1-8 ArwesSounds(Frame)-border-0-1-48 ArwesSounds(Frame)-borderTop-0-1-11"
          />
          <div
            class="ArwesSounds(Frame)-border-0-1-8 ArwesSounds(Frame)-border-0-1-48 ArwesSounds(Frame)-borderBottom-0-1-12"
          />
          <div
            class="ArwesSounds(Frame)-corner-0-1-13 ArwesSounds(Frame)-corner-0-1-49 ArwesSounds(Frame)-cornerLT-0-1-14 ArwesSounds(Frame)-cornerLT-0-1-50"
          />
          <div
            class="ArwesSounds(Frame)-corner-0-1-13 ArwesSounds(Frame)-corner-0-1-49 ArwesSounds(Frame)-cornerLB-0-1-15 ArwesSounds(Frame)-cornerLB-0-1-51"
          />
          <div
            class="ArwesSounds(Frame)-corner-0-1-13 ArwesSounds(Frame)-corner-0-1-49 ArwesSounds(Frame)-cornerRT-0-1-16 ArwesSounds(Frame)-cornerRT-0-1-52"
          />
          <div
            class="ArwesSounds(Frame)-corner-0-1-13 ArwesSounds(Frame)-corner-0-1-49 ArwesSounds(Frame)-cornerRB-0-1-17 ArwesSounds(Frame)-cornerRB-0-1-53"
          />
          <div
            class="ArwesSounds(Frame)-box-0-1-6 ArwesSounds(Frame)-box-0-1-47"
          >
            <div
              class="ArwesSounds(Frame)-children-0-1-7"
            >
              <div
                class="Highlight-root-0-1-31"
              >
                <div
                  class="Highlight-children-0-1-33"
                >
                  <button
                    class="ArwesSounds(Button)-button-0-1-3 ArwesSounds(Button)-button-0-1-45"
                  >
                    9x9
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
